docs/DebuggingTheCompiler.md

```diff
@@ -29,49 +23,63 @@ The most important thing when debugging the compiler is to examine the IR.
 Here is how to dump the IR after the main phases of the Swift compiler
 (assuming you are compiling with optimizations enabled):
 
-#. **Parser**. To print the AST after parsing::
+* **Parser** To print the AST after parsing::
 
-    swiftc -dump-ast -O file.swift
+```bash
+swiftc -dump-ast -O file.swift
+```
 
-#. **SILGen**. To print the SIL immediately after SILGen::
+* **SILGen** To print the SIL immediately after SILGen::
 
-    swiftc -emit-silgen -O file.swift
+```bash
+swiftc -emit-silgen -O file.swift
+```
 
-#. **Mandatory SIL passes**. To print the SIL after the mandatory passes::
+* **Mandatory SIL passes** To print the SIL after the mandatory passes::
 
-    swiftc -emit-sil -Onone file.swift
+```bash
+swiftc -emit-sil -Onone file.swift
+```
 
   Well, this is not quite true, because the compiler is running some passes
   for -Onone after the mandatory passes, too. But for most purposes you will
   get what you want to see.
 
-#. **Performance SIL passes**. To print the SIL after the complete SIL
+* **Performance SIL passes** To print the SIL after the complete SIL
    optimization pipeline::
 
-    swiftc -emit-sil -O file.swift
+```bash
+swiftc -emit-sil -O file.swift
+```
 
-#. **IRGen**. To print the LLVM IR after IR generation::
+* **IRGen** To print the LLVM IR after IR generation::
 
-    swiftc -emit-ir -Xfrontend -disable-llvm-optzns -O file.swift
+```bash
+swiftc -emit-ir -Xfrontend -disable-llvm-optzns -O file.swift
+```
 
-4. **LLVM passes**. To print the LLVM IR after LLVM passes::
+* **LLVM passes** To print the LLVM IR after LLVM passes::
 
-    swiftc -emit-ir -O file.swift
+```bash
+swiftc -emit-ir -O file.swift
+```
 
-5. **Code generation**. To print the final generated code::
+* **Code generation** To print the final generated code::
 
-    swiftc -S -O file.swift
+```bash
+swiftc -S -O file.swift
+```
 
 Compilation stops at the phase where you print the output. So if you want to
 print the SIL *and* the LLVM IR, you have to run the compiler twice.
-The output of all these dump options (except ``-dump-ast``) can be redirected
-with an additional ``-o <file>`` option.
+The output of all these dump options (except `-dump-ast`) can be redirected
+with an additional `-o <file>` option.
 
 Debugging Diagnostic Emission
 -----------------------------
 
 Asserting on first emitted Warning/Assert Diagnostic
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+----------------------------------------------------
```

Comment:
Each new style of section headers in reStructuredText creates another section level.

In this document:

* H1: `===`
* H2: `---`
* H3: `~~~` (i.e. `###` in Markdown)